torch/csrc/inductor/aoti_runner/model_container_runner.h

```diff
@@ -22,11 +22,13 @@ class TORCH_API AOTIModelContainerRunner {
       delete;
   AOTIModelContainerRunner& operator=(AOTIModelContainerRunner&& other) =
       delete;
-  ~AOTIModelContainerRunner();
+  virtual ~AOTIModelContainerRunner();
 
-  std::vector<at::Tensor> run(
+  virtual std::vector<at::Tensor> run(const std::vector<at::Tensor>& inputs);
```

Comment:
It changes the API definition. According to the issue description, the `null` stream means the current stream for CUDA. The same semantic should also be applicable for Intel GPU. Since this PR has fixed the virtual function routine, the runner instance could dispatch `run` function to the right device container runner.

Reply:
This relates to our definition of the run API. At first I thought it would be more generic to define it as `std::vector<at::Tensor> run(const std::vector<at::Tensor>& inputs, AOTInductorStreamHandle stream_handle = nullptr);`
. But this means that in `model_container_runner_xpu.h`, `model_container_runner_xpu.h`, `model_container_runner_cpu.h`,  we need to define this override API as well. Since `AOTIModelContainerRunnerCpu/Xpu/Cuda` will be used in pybind,  we'll get compilation error: 
```
torch/csrc/inductor/aoti_runner/pybind.cpp:21:11: required from here
/usr/include/c++/11/type_traits:1422:38: error: invalid use of incomplete type ‘struct AOTInductorStreamOpaque’
1422 | : public integral_constant<bool, __is_base_of(_Base, _Derived)>
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/xinanlin/xinanlin/pytorch/torch/csrc/inductor/aoti_runner/model_container_runner.h:5,
from torch/csrc/inductor/aoti_runner/model_container_runner_cpu.h:4,
from torch/csrc/inductor/aoti_runner/pybind.cpp:1:
torch/csrc/inductor/aoti_runtime/interface.h:16:8: note: forward declaration of ‘struct AOTInductorStreamOpaque’
```
The current design of this PR defines the interface of cpu/xpu/cuda container runner as `std::vector<at::Tensor> run(const std::vector<at::Tensor>& inputs);` is just keep the same before PR, so I think there is no issue to land it.
 
And if we prefer the interface `run` as `std::vector<at::Tensor> run(const std::vector<at::Tensor>& inputs, AOTInductorStreamHandle stream_handle = nullptr);` ,  we can create an adapter between pybind interface and the AOTI runner container for the container to resolve the above compilation issue in a new PR.

Reply:
Agree with @EikanWang , this is an API breaking change. I think if you declare `stream_handle` as `void*`, it should solve your compilation issue.

Reply:
Hi, @desertfire I've updated the PR, could you please have a review? BTW could you please also review this series of PRs? They are all for AOTi XPU and are ready for review.